src/sql/src/plan/statement/show.rs

```diff
@@ -722,6 +740,34 @@ impl<'a> ShowSelect<'a> {
     }
 }
 
+pub struct ShowColumnsSelect<'a> {
+    id: GlobalId,
+    new_resolved_ids: ResolvedIds,
+    show_select: ShowSelect<'a>,
+}
+
+impl<'a> ShowColumnsSelect<'a> {
+    pub fn describe(self) -> Result<StatementDesc, PlanError> {
+        self.show_select.describe()
+    }
+
+    pub fn plan(self) -> Result<Plan, PlanError> {
+        let select_plan = self.show_select.plan()?;
+        match select_plan {
+            Plan::Select(select_plan) => Ok(Plan::ShowColumns(ShowColumnsPlan {
+                id: self.id,
+                select_plan,
+                new_resolved_ids: self.new_resolved_ids,
+            })),
+            _ => Err(PlanError::Unstructured(format!("SHOW COLUMNS didn't produce a SelectPlan as expected, produced {:?} instead. Please file a bug.", select_plan)))
```

Comment:
I think we'd end up showing this string to the user, can you instead use a little more generic message like "SHOW COLUMNS produced an unexpected plan", and then include the more descriptive info in a `tracing::error!`?